src/main.rs

```diff
@@ -215,23 +221,18 @@ async fn handle_request(
     let bytes = (hyper::body::to_bytes(body).await.context(ChunkingRequest)?).to_vec();
 
     let path_and_query = match uri.path_and_query() {
-        Some(v) => v.as_str().replace(&api_url, "").into(),
+        Some(v) => v.as_str().replace(&api_url, ""),
         None => {
             debug!("No path in URI: {:?}", uri);
 
             return Err(RequestError::NoPath { uri });
         }
     };
-    let body = if bytes.is_empty() { None } else { Some(bytes) };
     let p = path_name(&path);
-    let raw_request = TwilightRequest {
-        body,
-        form: None,
-        headers: Some(headers),
-        method,
-        path,
-        path_str: path_and_query,
-    };
+    let raw_request = RequestBuilder::raw(method, path, path_and_query)
+        .body(bytes)
+        .headers(headers.into_iter().map(|(k, v)| (k.unwrap(), v)))
```

Comment:
I am not 100% sure what to do here. Should we return an error if there are two headers with the same name? (which is the only case when the key would be None)

Reply:
We could probably just take the first. I can't think of a reason to send more than one, and IMO it's less of an error worth failing the request and more of a "why are you doing this? oh well."

Reply:
Alright, applied that